lib/presentation/pages/registration_page/registration_page.dart

```diff
@@ -1,67 +1,120 @@
 import 'package:auto_route/auto_route.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:trip_tally/presentation/utils/enums/context_extensions.dart';
+import 'package:trip_tally/presentation/utils/enums/errors.dart';
 import 'package:trip_tally/presentation/utils/router/app_router.dart';
 
+import '../../../injectable/injectable.dart';
 import '../../theme/app_dimensions.dart';
 import '../../widgets/custom_floating_action_button.dart';
+import '../../widgets/custom_snack_bar.dart';
 import '../../widgets/custom_text_field.dart';
 import '../../widgets/welcome_subtitle.dart';
 import '../../widgets/welcome_text.dart';
+import 'bloc/registration_bloc.dart';
 
 @RoutePage()
 class RegistrationPage extends StatelessWidget {
-  const RegistrationPage({super.key});
+  const RegistrationPage({super.key, @visibleForTesting this.bloc});
+
+  final RegistrationBloc? bloc;
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (context) => bloc ?? getIt<RegistrationBloc>(),
+      child: BlocConsumer<RegistrationBloc, RegistrationState>(
+        listener: (context, state) => state.maybeWhen(
+          orElse: () => const SizedBox.shrink(),
+          loading: () => const Center(
+            child: CircularProgressIndicator(),
+          ),
+          failure: (error) => customSnackBar(
+            context,
+            error.errorText(context),
+          ),
+          success: () => customSnackBar(context, 'Zostałeś zarejestrowany'), //TODO: push to HomePage
+        ),
+        builder: (context, state) => state.maybeWhen(
+          orElse: () => const SizedBox.shrink(),
+          failure: (_) => _Body(),
+          success: () => _Body(),
+          initial: () => _Body(),
+          loading: () => Container(color: Colors.white),
```

Comment:
You repeating the same widget multiple times maybe:
orElse: () => _Body(),
loading: () => CircularProgress